### PR TITLE
[Fix] : Admin feature visibility bug

### LINKED ISSUE #4581: Admin Only feature visible on Edit Profile Page
An Admin-only feature is visible to users  on the Edit profile page
### Evidence / Screenshot (if possible)
image

### Relevant url?

https://openlibrary.org/people/{username}?m=edit
### Steps to Reproduce

1. Go to My profile
2. Click Edit

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Chrome
- **Operating system?** Windows
- **Environment (prod/dev/local)?** prod

### PR DESCRIPTION
**Pull request by infern018**
Closes #4581 

### Technical

In the edit.html file of the user section ```$:macros.TypeChanger(page.type)``` was placed unnecessarily, so I commented it out and now that section is not visible on the edit page.

### Screenshot

BEFORE:
Screenshot (372)_LI
.
.
.
AFTER:
Screenshot (373)

### Stakeholders
@jamesachamp @jdlrobson

### CODE REVIEW

**Review by Yashs911 — CHANGES_REQUESTED** [review]
@infern018 Actually this feature is admin-specific and removing it completely might not be the right solution. Try to look at the TypeChanger Macros and the CSS related to that feature.
image

**Comment by infern018** [comment]
@Yashs911  can u tell me how can  I log in as admin/or see such, because I will need to verify the changes on my local setup too...

**Comment by Yashs911** [comment]
> @Yashs911 can u tell me how can I log in as admin/or see such, because I will need to verify the changes on my local setup too...

Follow this https://github.com/internetarchive/openlibrary/blob/master/CONTRIBUTING.md#user-content-logging-in-as-admin
but even after clearing my cookies, I am unable to login as admin

**Comment by infern018** [comment]
I've changed the CSS of **pageType** block to 
`display:none` when user is not logged as admin, 
and to,
`display:block` when user is logged as an admin

opl2
opl1

**Comment by infern018** [comment]
if there are any changes to be done please let me know @jamesachamp @jdlrobson

**Comment by mekarpeles** [comment]
I'll pull the change on to dev or staging and test, thank you @infern018 and great feedback @Yashs911 !

File: openlibrary/macros/TypeChanger.html [hunk 1]
```diff
@@ -3,6 +3,11 @@
 $# and any generic page in the admin UI; e.g. creating a page.
 <div class="formElement pagetype collapse adminOnly" id="pageType">
     $if ctx.user and ctx.user.is_admin():
+        <style>
```
Comment (jdlrobson) [inline openlibrary/macros/TypeChanger.html:6]: Rather than add an inline style please add this in one of the stylesheets. There is a big effort underway at the moment for us to move away from inline styles and scripts.

To do this we'll need to add a body class in openlibrary/templates/site/body.html or using JavaScript

e.g. 

```
#pageType { display: none;}
.client-admin #pageType {display : block;}
```
We should make sure we don't ever cache the client-admin class for anonymous page views.

File: openlibrary/macros/TypeChanger.html [hunk 2]
```diff
@@ -22,4 +27,9 @@
         </div>
     $else:
         <input type="hidden" name="type.key" value="$type"/>
+        <style>
```
Comment (jdlrobson) [inline openlibrary/macros/TypeChanger.html:30]: see coment above

File: openlibrary/templates/type/user/edit.html [hunk 3]
```diff
@@ -42,6 +42,7 @@ <h1>
 
         $:macros.TypeChanger(page.type)
 
+
```
Comment (jdlrobson) [inline openlibrary/templates/type/user/edit.html:45]: appears unrelated to proposed change. Please remove.

**Review by jdlrobson — CHANGES_REQUESTED** [review]

File: openlibrary/catalog/marc/tests/test_data/xml_input/39002054008678.yale.edu_marc.xml [hunk 4]
```diff
@@ -1 +1 @@
...
... /</marc:subfield><marc:subfield code="c">by Thomas Conant.</marc:subfield></marc:datafield><marc:datafield tag="260" ind1=" " ind2=" "><marc:subfield code="a">Toronto :</marc:subfield><marc:subfield code="b">W. Briggs,</marc:subfield><marc:subfield code="c">1898.</marc:subfield></marc:datafield><marc:datafield tag="300" ind1=" " ind2=" "><marc:subfield code="a">243 p., [27] leaves of plates :</marc:subfield><marc:subfield code="b">ill., map, port. ;</marc:subfield><marc:subfield code="c">24 cm.</marc:subfield></marc:datafield><marc:datafield tag="651" ind1=" " ind2="0"><marc:subfield code="a">Ontario</marc:subfield><marc:subfield code="x">History.</marc:subfield></marc:datafield><marc:datafield tag="651" ind1=" " ind2="0"><marc:subfield code="a">Ontario</marc:subfield><marc:subfield code="x">Description and travel.</marc:subfield></marc:datafield><marc:datafield tag="927" ind1=" " ind2=" "><marc:subfield code="a">20000807R </marc:subfield></marc:datafield><marc:datafield tag="928" ind1=" " ind2=" "><marc:subfield code="a">ACOC0600</marc:subfield></marc:datafield><marc:datafield tag="948" ind1=" " ind2=" "><marc:subfield code="a">RECON</marc:subfield></marc:datafield></marc:record>
```
Comment (cclauss) [inline openlibrary/catalog/marc/tests/test_data/xml_input/39002054008678.yale.edu_marc.xml:1]: What was changed in this test file?  The diff does not highlight any change.

If you add one and only one carriage return (\n) to the end of the file the GitHub will remove the ⛔.

**Comment by Yashs911** [comment]
@infern018 Any update on this?

**Comment by mekarpeles** [comment]
Moving to a draft:
1. the MARC changes should be removed, are not relevant to this PR
2. The styles should be moved to css, not inline

**Comment by cdrini** [comment]
Closing as stale; feel free to re-open when you get some time, @infern018 !
